### PR TITLE
Monkey patch bundler EndpointSpecification

### PR DESCRIPTION
**Pull request by pavera**
We have had a number of customer escalations reporting timeouts and excessive requests to private gem servers.

This is a known performance regression in bundler documented here: https://github.com/rubygems/rubygems/issues/5385

Through discussions with @deivid-rodriguez he is working on an upstream fix, but in the meantime we can patch the EndpointSpecification class so that it no longer retrieves gemspec.rz files from private servers.

I've confirmed through manual tests against a private gem server and a repo that consumes said registry that this patch greatly improves performance (~10 minutes to run dry-run.rb down to 30-45s)

### CODE REVIEW

File: bundler/helpers/v2/lib/functions.rb [hunk 1]
```diff
@@ -143,7 +143,7 @@ def self.set_bundler_flags_and_credentials(dir:, credentials:)
 
       Bundler.settings.set_command_option(
         cred.fetch("host"),
-        token.gsub("@", "%40F").gsub("?", "%3F")
+        token.gsub("@", "%40").gsub("?", "%3F")
```
Comment (deivid-rodriguez) [inline bundler/helpers/v2/lib/functions.rb:146]: Is this related to the fix?

**Review by deivid-rodriguez — COMMENTED** [review]
The patch looks perfect! 👍 Just a question about a part of the commit that did not seem related.

File: bundler/helpers/v2/lib/functions.rb (hunk 1 above)
Comment (pavera) [reply bundler/helpers/v2/lib/functions.rb:146]: Only in that testing private gem server access via dry-run.rb was broken because of that spurious F.

Comment (deivid-rodriguez) [reply bundler/helpers/v2/lib/functions.rb:146]: I see, so this bug does not affect end users at all, only `dry-run.rb`, correct?

Comment (pavera) [reply bundler/helpers/v2/lib/functions.rb:146]: Yes.

**Review by deivid-rodriguez — APPROVED** [review]
